proposal/color-4-new-spaces.md

```diff
@@ -0,0 +1,1790 @@
+    > provide additional options in the future.
+
+  * Let `space` be the value of `$space`.
+
+* Return the result of calling
+  `color.adjust($color, $hue: 180deg, $space: space)`.
+
+### `color.invert()`
+
+```
+invert($color, $space: null)
+```
+
+This function is also available as a global function named `invert()`.
+
+* If `$color` is not a color, throw an error.
+
+* If `$space` is null:
+
+  * If `$color` is a legacy color, let `space` be `rgb`, and let `mix-space`
+    be null.
+
+    > This allows us to also enforce legacy behavior in the final weighted mix.
+
+  * Otherwise, throw an error.
+
+* Otherwise:
+
+  * If `$space` is not a [known color space], throw an error.
+
+  * Let `space` be `$space`, and let `mix-space` be `$space`.
+
+* Let `color` be the result of [converting] and [gamut mapping] `$color`
+  into the color space `space`.
+
+* If `space == hwb`:
+
+  * Let `hue`, `whiteness`, and `blackness` be the three elements of `color`'s
+    channels.
+
+  * Let `hue-out` be the result of `(hue + 180deg) % 360deg`.
+
+  * Let `gray` be the result of `whiteness + blackness`.
+
+  * Let `white` be `gray - whiteness`, and let `black` be `gray - blackness`.
+
+  * Let `invert` be the result of calling
+    `color.change(color, $hue: hue-out, $whiteness: white, $blackness: black)`.
+
+* Otherwise:
+
+  * Let `invert` be the value of `color`.
+
+  * For each `channel` element in `color`'s channels:
+
+    * If `channel` represents a polar-angle hue:
+
+      * Let `new` be `(channel + 180deg) % 360deg`.
+
+    * Otherwise, if `channel` represents either chroma or saturation:
+
+      * Let `new` be `channel`.
+
+    * Otherwise:
+
+      * Let `min` and `max` be the minimum and maximum values defined for
+        `channel` in `space`.
+
+      * Let `new` be `max - channel` if `min == 0`, and `channel * -1` otherwise.
+
+    * Set the corresponding channel of `invert` to be `new`.
+
+* Return the result of calling `color.mix(invert, color, $weight, mix-space)`.
+
+### `color.grayscale()`
+
+```
+grayscale($color)
+```
+
+> No space argument is provided, since the results should always be in gamut.
+
+This function is also available as a global function named `grayscale()`.
+
+* If `$color` is not a color, throw an error.
+
+* If `$color` is a legacy color:
+
+  * Return the result of [converting] `$color` to `hsl`, and changing the
+    'saturation' channel to 0.
+
+* Otherwise:
+
+  * Let `origin` be `$color`'s [known color space].
+
+  * Let `color` be the result of [converting] `$color` to `oklch`, and
+    setting the `chroma` channel to 0.
+
+  * Return the result of [converting] `color` to `origin`.
+
+### `color.ie-hex-str()`
+
+This function is also available as a global function named `ie-hex-str()`. Both
+functions are deprecated.
+
+```
+ie-hex-str($color)
+```
+
+* If `$color` is not a color, throw an error.
+
+* Let `rgb` be the result of [converting] `$color` to `rgb`.
```

Comment:
I think this should gamut-map too, since there's no way to represent out-of-gamut colors in the 8-digit hex format.